src/converters/RpcHttpConverters.ts

```diff
@@ -55,35 +55,40 @@ export function fromNullableMapping(
  * 'http' types are a special case from other 'ITypedData' types, which come from primitive types.
  * @param inputMessage  An HTTP response object
  */
-export function toRpcHttp(inputMessage): RpcTypedData {
+export function toRpcHttp(inputMessage: HttpResponse): RpcTypedData {
```

Comment:
The user could pass us literally anything as the inputMessage so I'd rather be a bit more careful and use `unknown` as the type until after we've validated it on line 60. (Before your change, the implicit type was `any` which is the worst possible type. `HttpResponse` is somewhat better, but `unknown` is best in this case) 

Something like this:
```typescript
export function toRpcHttp(data: unknown): RpcTypedData {
    // Check if we will fail to find any of these
    if (typeof data !== 'object' || Array.isArray(data)) {
        throw new AzFuncSystemError(
            "The HTTP response must be an 'object' type that can include properties such as 'body', 'status', and 'headers'. Learn more: https://go.microsoft.com/fwlink/?linkid=2112563"
        );
    }
    const inputMessage: HttpResponse = data || {};
```

Reply:
@ejizba As far as  I understand, this function isn't called directly by user code, but rather our code calls it using user input. Isn't it better validation to enforce the `HttpResponse` type on the argument so that TypeScript throws an error if our code tries to pass user input directly without ensuring it's of an `HttpResponse` type? I'm saying this because the validation that this function does (ensuring it's an object) isn't enough to guarantee it's an `HttpResponse`.

Reply:
>> Isn't it better validation to enforce the HttpResponse type on the argument so that TypeScript throws an error if our code tries to pass user input directly without ensuring it's of an HttpResponse type?

> For TypeScript users this is true, but a lot of our users (probably the majority?) are JavaScript users so a TypeScript type doesn't enforce anything at all. That's why we need to do both.

But our code is the only one that uses this function, and our code is only TypeScript, no?

Reply:
Our code is the only one that uses this function, yes. There are several layers up the stack that are using `any` and should be using `unknown`. I'm not asking you to fix the whole call stack because that would be a pain - but eventually you would get to a point where it's closer to user code

Reply:
> The v3.x branch uses `any` waaayyy too much which means we lose a lot of type validation. I've mostly fixed it on v4.x, but if everything was properly typed this wouldn't even be a discussion

Are you saying that, because our code uses `any` so much, the fact that we don't get any build errors when saying `data: HttpResponse` doesn't guarantee that we are actually always passing an `HttpResponse` to this function, but that there is some validation missing, because `any` is used? Am I understanding this correctly?

Reply:
Yes there is some validation missing

Reply:
That makes sense, thanks!